packages/media-utils/src/utils/validate-mime-type.ts

```diff
@@ -40,4 +45,18 @@ export function validateMimeType( file: File, allowedTypes?: string[] ) {
 			file,
 		} );
 	}
+
+	if ( typesNotSupportedByServer && typesNotSupportedByServer[ file.type ] ) {
```

Comment:
By the way, why isn't `typesNotSupportedByServer` just a simple array?

Would the value ever be `false`? One can just remove an item from the array if wanted.

Reply:
Good point, this is a holdover from the original approach with an array of types and their support status.

I will update this.